dtfs-central-api/src/v1/controllers/tfm/amendments/tfm-get-amendments.controller.js

```diff
@@ -0,0 +1,346 @@
+const { ObjectId } = require('mongodb');
+const db = require('../../../../drivers/db-client');
+const CONSTANTS = require('../../../../constants');
+
+/* returns an array of object containing all amendments in progress
+ * [{
+ *    "amendmentId": "62692866ce546902bfcd9168",
+ *    "createdAt": 1651058790,
+ *    "updatedAt": 1651059653,
+ *    "status": "In progress",
+ * }]
+ */
+const findAllAmendmentsByStatus = async (status) => {
+  try {
+    const collection = await db.getCollection('tfm-facilities');
+    const amendment = await collection.aggregate([
+      { $project: { _id: 0, amendments: '$amendments' } },
+      { $unwind: '$amendments' },
+      { $match: { 'amendments.status': status } },
+      { $group: { _id: '$_id', amendments: { $push: '$amendments' } } },
+      { $project: { _id: 0, amendments: 1 } },
+    ]).toArray();
+
+    // returns the amendment object for the given facilityId and amendmentId
+    return amendment[0]?.amendments ?? null;
+  } catch (err) {
+    console.error('Unable to find amendments object %O', { err });
+    return null;
+  }
+};
+
+exports.getAllAmendmentsInProgress = async (req, res) => {
+  const amendment = await findAllAmendmentsByStatus(CONSTANTS.AMENDMENT.AMENDMENT_STATUS.IN_PROGRESS) ?? [];
+
+  return res.status(200).send(amendment);
+};
+
+/* returns an array of object containing all properties for a given facilityId:
+ * [{
+ *    "amendmentId": "62692866ce546902bfcd9168",
+ *    "createdAt": 1651058790,
+ *    "updatedAt": 1651059653,
+ *    "status": "In progress",
+ * }]
+ */
+const findAllAmendmentsByFacilityId = async (facilityId) => {
+  try {
+    const collection = await db.getCollection('tfm-facilities');
+    const amendment = await collection.aggregate([
+      { $match: { _id: ObjectId(facilityId) } },
+      { $project: { _id: 0, amendments: '$amendments' } },
+    ]).toArray();
+
+    // returns the amendment object for the given facilityId and amendmentId
+    return amendment[0]?.amendments ?? null;
```

Comment:
Index check please for `0` here and below please.

Reply:
we have the `?` flag to handle that